Implement `append`, `prepend`, and `concatenate` (#4792)
tenzir/tenzir: web/docs/tql2/functions/concatenate.md

# concatenate

Merges two lists.

```tql

> concatenate(list1:list, list2:list) -> list

Reviewer: ```suggestion
concatenate(xs:list, ys:list) -> list
```

```

## Description

> The `concatenate` function returns a list containing all elements from `list1`
> and `list2` in order. `concatenate(xs, ys)` is equivalent to `[...xs, ...ys]`.

Reviewer: ```suggestion
The `concatenate` function returns a list containing all elements from `xs`
and `ys` in order. The expression `concatenate(xs, ys)` is equivalent to `[...xs, ...ys]`.
```

## Examples

### Concatenate two lists

```tql
from {xs: [1, 2], ys: [3, 4]}
zs = concatenate(xs, ys)
```

```tql
{
  xs: [1, 2],
  ys: [3, 4],
  zs: [1, 2, 3, 4]
}
```

## See Also

[`append`](append.md), [`prepend`](prepend.md)

> 

Reviewer: ```suggestion
```
